disperser/apiserver/server.go

```diff
@@ -315,84 +311,103 @@ func (s *DispersalServer) getAccountRate(origin, authenticatedAddress string, qu
 
 }
 
-func (s *DispersalServer) checkRateLimitsAndAddRates(ctx context.Context, blob *core.Blob, origin, authenticatedAddress string) error {
+type limiterInfo struct {
+	Name     string
+	QuorumID core.QuorumID
+}
 
-	// TODO(robert): Remove these locks once we have resolved ratelimiting approach
-	s.mu.Lock()
-	defer s.mu.Unlock()
+func (s *DispersalServer) checkRateLimitsAndAddRatesToHeader(ctx context.Context, blob *core.Blob, origin, authenticatedAddress string) (bool, string, error) {
+
+	requestParams := make([]common.RequestParams, 0)
+
+	blobSize := len(blob.Data)
+	length := encoding.GetBlobLength(uint(blobSize))
 
 	for i, param := range blob.RequestHeader.SecurityParams {
 
-		rates, ok := s.rateConfig.QuorumRateInfos[param.QuorumID]
-		quorumId := string(param.QuorumID)
+		globalRates, ok := s.rateConfig.QuorumRateInfos[param.QuorumID]
 		if !ok {
-			return fmt.Errorf("no configured rate exists for quorum %d", param.QuorumID)
+			return false, "", fmt.Errorf("no configured rate exists for quorum %d", param.QuorumID)
 		}
+
 		accountRates, accountKey, err := s.getAccountRate(origin, authenticatedAddress, param.QuorumID)
 		if err != nil {
-			return err
+			return false, "", err
 		}
 
+		// Update the quorum rate
+		blob.RequestHeader.SecurityParams[i].QuorumRate = accountRates.Throughput
```

Comment:
for my understanding, how is this used downstream?

Reply:
Added a comment to the function definition describing this.